internal/pkg/keeper/conversion.go

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package keeper
+
+import (
+	"strconv"
+
+	"github.com/edgexfoundry/go-mod-configuration/v2/internal/pkg/keeper/api"
+)
+
+type pair struct {
+	Key   string
+	Value string
+}
+
+func convertMapToKVPairs(path string, interfaceMap interface{}) []*pair {
+	pairs := make([]*pair, 0)
+
+	pathPre := ""
+	if path != "" {
+		pathPre = path + api.KeyDelimiter
+	}
+
+	switch value := interfaceMap.(type) {
+	case []interface{}:
+		for index, item := range value {
+			nextPairs := convertMapToKVPairs(pathPre+strconv.Itoa(index), item)
+			pairs = append(pairs, nextPairs...)
+		}
+	case map[string]interface{}:
+		for index, item := range value {
+			nextPairs := convertMapToKVPairs(pathPre+index, item)
+			pairs = append(pairs, nextPairs...)
+		}
+	case int:
+		pairs = append(pairs, &pair{Key: path, Value: strconv.Itoa(value)})
+	case int64:
+		var value64 = int(value)
+		pairs = append(pairs, &pair{Key: path, Value: strconv.Itoa(value64)})
+	case float64:
+		pairs = append(pairs, &pair{Key: path, Value: strconv.FormatFloat(value, 'f', -1, 64)})
+	case bool:
+		pairs = append(pairs, &pair{Key: path, Value: strconv.FormatBool(value)})
+	case nil:
+		pairs = append(pairs, &pair{Key: path, Value: ""})
+	default:
+		pairs = append(pairs, &pair{Key: path, Value: value.(string)})
```

Comment:
sorry, my bad it should not go this way
https://medium.com/swlh/bad-go-frivolous-sprintf-2ad28fedf1a0

Reply:
the default should use `fmt.printf("%v", value)` anyway because don't convert the type without error checking
also, please add all the possible type for the future, such as float32, int8, and so on.

Reply:
Done